drivers/can/can_mcp251xfd.c

```diff
@@ -208,7 +208,7 @@ static int mcp251xfd_fifo_write(const struct device *dev, int mailbox_idx,
 
 static void mcp251xfd_rxobj_to_canframe(struct mcp251xfd_rxobj *src, struct can_frame *dst)
 {
-	memset(dst, 0, offsetof(struct can_frame, data));
+	memset(dst, 0, sizeof(*dst));
```

Comment:
I'm not sure if we need this one either.. Imo the user shouldn't try to read the data outside the DLC.

Is the concern that RTR frames have a non-zero DLC? Maybe we should drop these invalid packets in the driver?

Reply:
The DLC of an RTR frame tells the recipient how much data to reply with. Please see the linked issue for the bug report.

Reply:
But in this case, if the RTR flag is set the user should know that the data is invalid. I don't see why we need to zero out data bytes.

Reply:
Otherwise we can leak uninitialized stack contents to the application/user mode thread.